.github/workflows/tests.yml

```diff
@@ -277,76 +286,74 @@ jobs:
       max-parallel: 10
     timeout-minutes: 120
     env:
-      pytest-replay: --replay-record-dir=pytest-replay/ --replay-base-name=macOS-${{ matrix.conda-version }}-Py${{ matrix.python-version }}
+      CONDA_CHANNEL_LABEL: ${{ matrix.conda-version == 'canary' && 'conda-canary/label/dev' || 'defaults' }}
+      REPLAY_NAME: macOS-${{ matrix.conda-version }}-Py${{ matrix.python-version }}
+      REPLAY_DIR: ${{ github.workspace }}/pytest-replay/
+      PYTEST_MARKER: ${{ matrix.test-type == 'serial' && 'serial' || 'not serial' }}
+      PYTEST_NUMPROCESSES: ${{ matrix.test-type == 'serial' && 0 || 'auto' }}
 
     steps:
       - name: Checkout repository
         uses: actions/checkout@v2
         with:
+          path: ./src/
           fetch-depth: 0
 
-      - name: Install miniconda
+      - name: Checkout test recipe
+        uses: actions/checkout@v2
+        with:
+          repository: conda/conda_build_test_recipe
+          path: ./conda_build_test_recipe/
+          fetch-depth: 0
+
+      - name: Setup miniconda
+        uses: conda-incubator/setup-miniconda@v2
+        with:
+          condarc-file: ./src/ci/github/.condarc
+          python-version: ${{ matrix.python-version }}
+
+      - name: Setup environment
         run: |
-          set -x -e -u
-          curl -o ${{ github.workspace }}/miniconda.sh https://repo.anaconda.com/miniconda/Miniconda3-latest-MacOSX-x86_64.sh
-          chmod +x ${{ github.workspace }}/miniconda.sh
-          ${{ github.workspace }}/miniconda.sh -b -p ${{ github.workspace }}/miniconda
-          source ci/github/activate_conda "${{ github.workspace }}/miniconda/bin/python"
+          conda install -q -y -c defaults --file tests/requirements.txt
+          conda install -q -y -c defaults --file tests/requirements-macos.txt
+          conda install -q -y ${{ env.CONDA_CHANNEL_LABEL }}::conda
```

Comment:
```suggestion
          conda install -q -y -c defaults \
            --file tests/requirements.txt \
            --file tests/requirements-macos.txt
          conda install -q -y -c defaults ${{ env.CONDA_CHANNEL_LABEL }}::conda
```
same as above

Reply:
same suggestion.. we could combine all the necessary reqs for dependency installation